File: timeoff.ui/src/app/site/settings/teams/team.model.ts
import { UserModel } from '@services/company/user.model';

export interface TeamModel {
    id: number;
    name: string;
    manager: UserModel;
    allowance: number;
    employeeCount: number;
    includePublicHolidays: boolean;
    isAccruedAllowance: boolean;
}

File: Application/Teams/TeamModel.cs
﻿namespace Timeoff.Application.Teams
{
    public abstract record TeamModel
    {
        public string Name { get; init; } = null!;
        public double Allowance { get; init; }
        public bool IncludePublicHolidays { get; init; }

        public bool IsAccruedAllowance { get; init; }
    }
}